ang/civicase/ContactCaseTab.js

```diff
@@ -81,6 +83,54 @@
       updateCase(caseListIndex, params);
     }
 
+    /*
+     * Watcher for civicase::contact-record-list::view-case event
+     *
+     * @params {Object} event
+     * @params {Object} cases of the list
+     */
+    function contactRecordListViewCaseWatcher (event, caseObj) {
+      setCaseAsSelected(caseObj);
+    }
+
+    /**
+     * Fetch additional information about the contacts
+     *
+     * @param {object} event
+     * @param {array} cases
+     */
+    function fetchContactsData (cases) {
+      var contacts = [];
+
+      _.each(cases, function (caseObj) {
+        contacts = contacts.concat(getAllContactIdsForCase(caseObj));
+      });
+
+      ContactsDataService.add(contacts);
+    }
+
+    /**
+     * Get all the contacts of the given case
+     *
+     * @param {object} caseObj
+     * @return {array}
+     */
+    function getAllContactIdsForCase (caseObj) {
+      var contacts = [];
+
+      _.each(caseObj.contacts, function (currentCase) {
+        contacts.push(currentCase.contact_id);
+      });
+
+      _.each(caseObj.activity_summary.next, function (activity) {
```

Comment:
When I tried passing `caseObj.next_activity` object to the activity card directive inside the details section it viewed fine, but when updating the values of the activity card in the edit activity model, it din't update the view. However, changing `caseObj.next_activity` to `caseObj.activity_summary.next[0]` fixed the issue. 

Also, since we need to use `caseObj.activity_summary.next` (which is an array) we need to iterate.

Please let me know if you need further clarifications.